server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1695,8 +1697,26 @@ public AliasFilter buildAliasFilter(ClusterState state, String index, Set<String
     /**
      * Returns a new {@link QueryRewriteContext} with the given {@code now} provider
      */
-    public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis) {
-        return new QueryRewriteContext(parserConfig, client, nowInMillis);
+    public QueryRewriteContext getRewriteContext(LongSupplier nowInMillis, IndicesRequest indicesRequest) {
+        Index[] indices = indexNameExpressionResolver.concreteIndices(clusterService.state(), indicesRequest);
```

Comment:
This may be a problem: `indexNameExpressionResolver.concreteIndices` fails on cross-cluster searches on this error: https://github.com/elastic/elasticsearch/blob/fc8e2b789730c01dd9e9d994b82a9a29beabaef7/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L1657

Any ideas for a resolution?

Reply:
I found a way to create new `IndicesOptions` with `ignore_unavailable` set to `true`, which is required to handle:

- indices on remote clusters
- invalid index names